modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -383,12 +396,12 @@ protected InboundPipeline getInboundPipeline(Channel ch, boolean isRemoteCluster
         );
     }
 
-    private static void addClosedExceptionLogger(Channel channel) {
-        Netty4Utils.addListener(channel.closeFuture(), channelFuture -> {
-            if (channelFuture.isSuccess() == false && logger.isDebugEnabled()) {
-                logger.debug(format("exception while closing channel: %s", channelFuture.channel()), channelFuture.cause());
+    private static void addClosedExceptionLogger(Netty4TcpChannel channel) {
+        channel.addCloseListener(ActionListener.wrap((ignored) -> {}, (e) -> {
```

Comment:
AIUI this is a change of behaviour right? `channel.closeFuture().isSuccess()` is never false so in practice this doesn't do anything today, whereas now we're going to log the exception that caused the channel to close. But I think we already log such exceptions, so maybe we don't need to duplicate that logging here.

Maybe we should just delete this dead code?

Also `ActionListener#wrap` isn't really necessary since the `onResponse` can't throw anything. If we do want to keep this behaviour it'd be better not to instantiate those two extra lambdas unnecessarily.